src/main/java/uk/gov/hmcts/ccd/data/definition/CachedCaseDefinitionRepository.java

```diff
@@ -90,7 +103,13 @@ public CaseTypeDefinitionVersion getLatestVersion(String caseTypeId) {
     @Override
     public JurisdictionDefinition getJurisdiction(String jurisdictionId) {
         LOGGER.debug("Will get jurisdiction '{}' from repository.", jurisdictionId);
-        return caseDefinitionRepository.getJurisdiction(jurisdictionId);
+        return jurisdictions.computeIfAbsent(jurisdictionId, caseDefinitionRepository::getJurisdiction);
+    }
+
+    @Override
+    public List<JurisdictionDefinition> getJurisdictions(List<String> jurisdictionIds) {
+        return jurisdictionsList.computeIfAbsent(prepareJurisdictionsKey(jurisdictionIds),
+            e -> caseDefinitionRepository.getJurisdictions(jurisdictionIds));
```

Comment:
This will fetch every list of jurisdictions separately even if they contain individual jurisdictions that were already fetched and cached. Not only performance will suffer, but also consistency of cached jurisdiction objects will be impacted. So, I believe we should maintain a single source of jurisdiction collection as a cache and serve any individual or list of jurisdictions from there.

Reply:
where do we use this method actually? since it's a request scope cache i don't think we actually gain anything unless same request flow calls this repeatedly. If we really want to cache then please follow Mutlu's suggestion of single source.

Reply:
Sure, I will make the respective changes